pkg/kube/controllers/extendedsecret/reconciler.go

```diff
@@ -73,50 +73,41 @@ func (r *ReconcileExtendedSecret) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// Check if secret was already generated
-	generatedSecret := &corev1.Secret{}
-	namespacedName := types.NamespacedName{
-		Namespace: instance.Namespace,
-		Name:      instance.Spec.SecretName,
+	// Check if secret could be generated when secret was already created
+	canBeGenerated, err := r.canBeGenerated(ctx, instance)
+	if err != nil {
+		ctxlog.Errorf(ctx, "Error reading the secret: %v", err.Error())
+		return reconcile.Result{}, err
 	}
-	err = r.client.Get(ctx, namespacedName, generatedSecret)
-	if err == nil {
-		ctxlog.Info(ctx, "Skip reconcile: secret already exists")
+	if !canBeGenerated {
+		ctxlog.Debugf(ctx, "SKip reconcile: secret '%s' could not be generated", instance.Spec.SecretName)
```

Comment:
typo in `SKip`.
Also please add more details in the log message, and make it `Info`.
`Skip reconcile: secret '%s' already exists and it's not generated`